.github/scripts/label_promoted_commits.py

```diff
@@ -44,49 +61,26 @@ def main():
-            }
-            headers = {
-                "Authorization": f"token {github_token}",
-                "Accept": "application/vnd.github.v3+json"
-            }
-            response = requests.get(search_url, headers=headers, params=params)
-            prs = response.json().get("items", [])
-            for pr in prs:
-                refs_match = re.findall(r'Refs (?:#|https.*?)(\d+)', pr["body"])
-                ref = re.search(r'branch-(\d+\.\d+)', args.ref)
-                if refs_match and ref:
-                    pr_number = int(refs_match[0])
-                    backport_pr_number = int(match.group(1))
-                    if pr_number in processed_prs:
-                        continue
-                    label_to_add = f'backport/{ref.group(1)}-done'
-                    label_to_remove = f'backport/{ref.group(1)}'
-                    pr = repo.get_pull(pr_number)
-                    current_labels = pr.get_labels()
-                    current_label_names = [label.name for label in current_labels]
-                    if label_to_remove in current_label_names:
-                        pr.remove_from_labels(label_to_remove)
-                    if label_to_add not in current_label_names:
-                        pr.add_to_labels(label_to_add)
-                    comment = f'Closed via {commit.sha}'
-                    add_comment_and_close_pr(backport_pr_number, repo, comment)
-                    processed_prs.add(pr_number)
-                else:
-                    assert re.match(r'/master', args.ref)
-                    pr_number = pr["number"]
-                    pr = repo.get_pull(pr_number)
-                    label_to_add = args.label
-                    current_labels = pr.get_labels()
-                    current_label_names = [label.name for label in current_labels]
-                    if label_to_add not in current_label_names:
-                        pr.add_to_labels(label_to_add)
-                processed_prs.add(pr_number)
+            pr_number = int(match.group(1))
+            pr = repo.get_pull(pr_number)
+            refs_pr = re.findall(r'Refs (?:#|https.*?)(\d+)', pr.body)
+            target_branch = re.search(r'branch-(\d+\.\d+)', args.ref)
+            if refs_pr and target_branch:
+                print(f'branch-{target_branch.group(1)}, pr number is: {pr_number}')
+                backport_pr_number = int(refs_pr[0])
+                if pr_number in processed_prs:
+                    continue
+                label_to_add = f'backport/{target_branch.group(1)}-done'
+                label_to_remove = f'backport/{target_branch.group(1)}'
+                switch_label_after_backport(repo, backport_pr_number, label_to_add, label_to_remove, commit.sha,
+                                            pr_number)
+            else:
+                print(f'master branch, pr number is: {pr_number}')
+                switch_label_after_backport(repo, pr_number, args.label)
```

Comment:
under what circumstances, we already have "promoted-to-master" label attached to a PR targeting `master`, and this PR is referenced with `Closes 1234` in a commit message when we are running this script and try to handle it?

IMHO, if we always attend PRs targeting master which do not have "promoted-to-master" label, we should just put
```py
if target_branch:
    # backport branch
else:
    # master branch
    pr = repo.get_pull(pr_number)
    pr.add_to_labels('promoted-to-master')
```

for a couple reasons:

- we don't need the flexibility of specifying a command line argument for this use case.
- this is more symmetric to how we handle the PRs targeting release branches, where we hardcode the label names to remove and to add. it's more explicit, and easier to understand.
- instead of using `switch_label_after_backport()`, it's much more straightforward. and reflect the intention better.